src/dbnode/storage/index/aggregate_results_entry_arraypool_gen.go

```diff
@@ -48,6 +48,20 @@ import (
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
+// AggregateResultsEntries is a slice of AggregateResultsEntry.
+type AggregateResultsEntries []AggregateResultsEntry
+
+// Size is the element size of the aggregated result entries.
+func (e AggregateResultsEntries) Size() int {
+	// NB: add 1 to the entries length for each entry's field.
+	length := len(e)
+	for _, entry := range e {
+		length += len(entry.Terms)
+	}
+
+	return length
+}
+
```

Comment:
This'll be reworked to not be inside a generated file (also refactor will help getting around having to do the full length calculation per incoming field)